test/support/coerceable_test_sqlserver.rb

```diff
@@ -19,21 +19,28 @@ def coerce_tests!(*methods)
         end
 
         def coerce_all_tests!
-          once = false
           instance_methods(false).each do |method|
             next unless method.to_s =~ /\Atest/
             undef_method(method)
-            once = true
           end
           STDOUT.puts "🙉 🙈 🙊  Undefined all tests: #{self.name}"
         end
 
         private
 
-        def coerced_test_warning(method)
-          method = instance_methods(false).select { |m| m =~ method } if method.is_a?(Regexp)
+        def coerced_test_warning(test_to_coerce)
+          if test_to_coerce.is_a?(Regexp)
+            method = instance_methods(false).select { |m| m =~ test_to_coerce }
+          else
+            method = test_to_coerce
+          end
+
           Array(method).each do |m|
-            result = undef_method(m) if m && method_defined?(m)
+            if m && method_defined?(m)
+              alias_method("original_#{test_to_coerce.inspect.tr('/\:"', '')}", m)
+              result = undef_method(m)
```

Comment:
I know ruby's lexical scope will define `result` for us but having it explicitly defined at the same level as it is used makes it easier to understand. Maybe something like this:

```ruby
result = if m && method_defined?(m)
           alias_method("original_#{test_to_coerce.inspect.tr('/\:"', '')}", m)
           undef_method(m)
         end
```

Reply:
Done